### PR TITLE
Migrate harvest action to use resources resources

### PR DESCRIPTION
**Pull request by dandehavilland**
PT https://www.pivotaltracker.com/story/show/171222777

---

Depends on artemis-ag/artemis_api#27

### CODE REVIEW

File: app/services/metrc_service/harvest.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def build_harvest_plants_payload(items, batch)
           ActualDate: harvest_date,
           Plant: item.relationships.dig('barcode', 'data', 'id'),
           Weight: average_weight,
-          UnitOfWeight: item.attributes['harvest_unit'],
```
Comment (dandehavilland) [inline app/services/metrc_service/harvest.rb]: @sd @betacar I'm trying to replace this with a reference to the `resource_unit` but there is no `resource_unit_id` for Item [(see items API response here)](https://github.com/artemis-ag/apps-cannapis/pull/22/files#diff-cf26d231da2367545c2f018b58a4f37cR1). Should this instead reference one of the `resources` in the harvest completion? If so, how do I resolve which one (in this example there are 4 `resources` with differing `resource_unit_id`s in the completion payload [(here)](https://github.com/artemis-ag/apps-cannapis/pull/22/files#diff-6344f85ce32f77eb9a55914162d989beR233)

Comment (betacar) [reply app/services/metrc_service/harvest.rb]: @dandehavilland For now, you'll have to request the facility's resource and look the one whose name has `wetweight` (@sd please correct me if my name nomenclature is wrong). That's the one you should use.

This is a temporary solution, while the API is improved for these cases.

Comment (dandehavilland) [reply app/services/metrc_service/harvest.rb]: OK, thanks @betacar. I have units by the names of `g of Wet Material - Boss Hog` and `g of Wet Material - 5th Element` so I'll filter based on `/Wet Material/`.

Comment (dandehavilland) [reply app/services/metrc_service/harvest.rb]: @betacar Having said that, I don't see either of those in a `harvest` completion or any of the associated `generate` completions.

Comment (sd) [reply app/services/metrc_service/harvest.rb]: Look for them in `process` completions out of `move` actions.

In general, we should look for both `process` and `generate` completions as triggers. Using one or the other will depend on the exact workflow for a customer (using separate stages between some of these phases or not)

File: Gemfile [hunk 2]
```diff
@@ -39,7 +39,9 @@ else
   gem 'Metrc', git: 'https://github.com/artemis-ag/Metrc.git' # rubocop:disable Bundler/DuplicatedGem
 end
 
-gem 'artemis_api', git: 'https://github.com/artemis-ag/artemis_api'
+gem 'artemis_api',
+    git: 'https://github.com/artemis-ag/artemis_api',
+    branch: 'feature/nested-completions'
```
Comment (betacar) [inline Gemfile]: This was merged. Can you please pull the new gem?

File: Gemfile.lock [hunk 3]
```diff
@@ -10,7 +10,8 @@ GIT
 
 GIT
   remote: https://github.com/artemis-ag/artemis_api
-  revision: 6d53c822d5197343329e143ed87754148e9ddeeb
+  revision: a78e7f6f5d796d7474248d7ae515b9b4713dce2d
+  branch: feature/nested-completions
```
Comment (betacar) [inline Gemfile.lock]: Same as above.

File: spec/support/token_store/artemis_token.json [hunk 4]
```diff
@@ -1 +1 @@
-{"access_token":"eyJhbGciOiJIUzUxMiJ9.eyJpc3MiOiJNeSBBcHAiLCJpYXQiOjE1ODE0OTQ5MDIsImp0aSI6ImIyMjQ3Yzk5LTU5NmUtNDBhMi1iNWZmLTI2ODhlY2Q2ZjVkYyIsInVzZXIiOnsiaWQiOjE2NzMsImVtYWlsIjoiZGFuQGRhbmRlaGF2aWxsYW5kLmNvbSJ9fQ.DIFjEl_fS11BUTq9EnxYTwFGXL2to79dDsmkDz4g5ZFSgVwAlqD_TcILwYz4sbeytiQWZL4a30MzLtaXvl8Gdw","expires_at":1581502102,"refresh_token":"XiozcSUW3PjUl6cpS2PQAbx3EoYHAWsVWGHxtQXt1k0"}
\ No newline at end of file
+{"access_token":"eyJhbGciOiJIUzUxMiJ9.eyJpc3MiOiJNeSBBcHAiLCJpYXQiOjE1ODI4OTk4NzAsImp0aSI6IjdiOWY3NDJlLTU5MjUtNGUxNS1iYmJlLTc3M2QyYjUzZWFhMyIsInVzZXIiOnsiaWQiOjExLCJlbWFpbCI6Im93bmVyQGNhbm5hZGVtby5jb20ifX0.vsV7QkjZUC2-EWv4it0Z_pbliCv3eqFlAL8tpvy2X_58CaJzMoEu_flJ3Zm1MJaNParvSBfcVIn9bTV4_206lA","expires_at":1582907070,"refresh_token":"kbr2kByzfo9j7-D67JyTt4XWrg886pgYLcQpKbJAHrk"}
```
Comment (betacar) [inline spec/support/token_store/artemis_token.json]: I believe we should implement a token generator. Let's leave it for later.

Comment (dandehavilland) [reply spec/support/token_store/artemis_token.json]: I've already added a rake task here: https://github.com/artemis-ag/apps-cannapis/blob/77a43d579f97eb8965ac9f738ff7d4dac0c96e44/lib/tasks/artemis_oauth.rake

Comment (betacar) [reply spec/support/token_store/artemis_token.json]: Awesome!

**Pull request merged**